src/routes/(app)/admin[[hash=admin_hash]]/reporting/+page.server.ts

```diff
@@ -2,25 +2,20 @@ import { collections } from '$lib/server/database';
 import { countryFromIp } from '$lib/server/geoip';
 import { sum } from '$lib/utils/sum';
 import { pojo } from '$lib/server/pojo.js';
-import { endOfMonth, startOfMonth } from 'date-fns';
 
 export async function load({ url }) {
-	const month = Number(url.searchParams.get('month'));
-	const year = Number(url.searchParams.get('year'));
-	let date = new Date();
-	if (year && month) {
-		date = new Date(year, month - 1);
-	}
+	const beginsDate = new Date(url.searchParams.get('beginsAt') || new Date());
```

Comment:
Begins at shoudl have a default value in the past, eg `subMonth(new Date(), 1)` 

Also you should use `zod` to validate the query params?